### DISCUSSION TITLE
Sandbox-Mode?

### DISCUSSION THREAD

**Discussion by Kinggrass** [abuzuhri/Amazon-SP-API-CSharp#10] [Q&A]
Is there a way to let all the requests go to the sandbox base url?

Edit:
A little bit of research in the repository  :) ....
Maybe I guess?:
EnvironemntManager.Environemnt = EnvironemntManager.Environments.Sandbox;

If that is the case, then environemnt is spelled with the e after the m and not before like environment.
So above has to be EnvironmentManager.Environment = EnvironmentManager.Environments.Sandbox;

**Comment by abuzuhri** [comment]
> Is there a way to let all the requests go to the sandbox base url?
> 
> Edit: A little bit of research in the repository :) .... Maybe I guess?: EnvironemntManager.Environemnt = EnvironemntManager.Environments.Sandbox;
> 
> If that is the case, then environemnt is spelled with the e after the m and not before like environment. So above has to be EnvironmentManager.Environment = EnvironmentManager.Environments.Sandbox;

Hi @Kinggrass 

I will try to fix in this weekend and for now not supporting 

/Tareq

**Reply by Kinggrass** [reply]
Hi @abuzuhri,

made a pull request for you which you can accept on the weekend for your ease ;).

**Comment by abuzuhri** [comment]
Hi @Kinggrass 

Pull request accepted and merge with Thanks :) .

**Reply by abuzuhri** [reply]
Environment url fixed now , but after test not work please double check from your side 


```csharp
    AmazonConnection amazonConnection = new AmazonConnection(new AmazonCredential()
    {
         AccessKey = "AKIAXXXXXXXXXXXXXXX",
         SecretKey = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
         RoleArn = "arn:aws:iam::XXXXXXXXXXXXX:role/XXXXXXXXXXXX",
         ClientId = "amzn1.application-XXX-client.XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
         ClientSecret = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
         RefreshToken= "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
         IsActiveLimitRate = true,
         Environment = Environments.Sandbox
     });

```

**Reply by Kinggrass** [reply]
Yes I am aware of it that it could not be working, because amazon is expecting explicit input on attributes for requests to the sandbox and unfortunately it does not match the type/format of production mode like in orders-api:

```
.
.
.
"x-amzn-api-sandbox": {
              "static": [
                {
                  "request": {
                    "parameters": {
                      "CreatedAfter": {
                        "value": "TEST_CASE_200"
                      },
                      "MarketplaceIds": {
                        "value": [
                          "ATVPDKIKX0DER"
                        ]
                    }
                  }
                },
               .
               .
               .
```
 and from the source code a DateTime Object is serialized, so it could not be fulfilled.
 Because obviously the DateTime Object does not produce "TEST_CASE_200".
 So for a Sandbox mode support some attributes have to be set static as a string during a request.
 ... alot of work to rewrite the code to support sandbox I know ^^

**Comment by abuzuhri** [comment]
Hi @Kinggrass 

Library now support sandbox

### Order List data from Sandbox
```CSharp
AmazonConnection amazonConnection = new AmazonConnection(new AmazonCredential()
{
     AccessKey = "AKIAXXXXXXXXXXXXXXX",
     SecretKey = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
     RoleArn = "arn:aws:iam::XXXXXXXXXXXXX:role/XXXXXXXXXXXX",
     ClientId = "amzn1.application-XXX-client.XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
     ClientSecret = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
     RefreshToken= "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
     Environment=Environments.Sandbox
});

var orders = amazonConnection.Orders.GetOrders
        (
         new FikaAmazonAPI.Parameter.Order.ParameterOrderList()
             {
              TestCase = Constants.TestCase200
             }
        );
```
